File: LeetCode75_LevelOne/Flood_Fill/Flood_Fill.cc
// 733. Flood Fill C++
#include <vector>
class Solution {
public:
    int rows, cols, to_change;
    std::vector<std::vector<int>> floodFill(std::vector<std::vector<int>>& image, int sr, int sc, int color) {
        rows = image.size();
        cols = image[0].size();
        to_change = image[sr][sc];
        fillNeigh(image, sr, sc, color);
        return image;
    }
    void fillNeigh(std::vector<std::vector<int>>& image, int sr, int sc, int color) {
        if (sr < 0 || sc < 0 || sr > rows - 1 || sc > cols - 1 || image[sr][sc] == color || image[sr][sc] != to_change) return;
        image[sr][sc] = color;
        fillNeigh(image, sr-1, sc, color);
        fillNeigh(image, sr+1, sc, color);
        fillNeigh(image, sr, sc-1, color);
        fillNeigh(image, sr, sc+1, color);
    }
};

File: Algorithm_LevelOne/Flood_Fill/Flood_Fill.py
# 733. Flood Fill Python

class Solution(object):
    def floodFill(self, image, sr, sc, color):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        self.fillNeigh(image, sr,sc,color, image[sr][sc], len(image), len(image[0]))
        return image
    def fillNeigh(self, image, r, c, color, to_change, rows, cols):
        if r < 0 or c < 0 or r > rows -1 or c > cols -1 or image[r][c] == color or image[r][c] != to_change:
            return
        image[r][c] = color
        self.fillNeigh(image, r-1, c, color, to_change, rows, cols)
        self.fillNeigh(image, r+1, c, color, to_change, rows, cols)
        self.fillNeigh(image, r, c-1, color, to_change, rows, cols)
        self.fillNeigh(image, r, c+1, color, to_change, rows, cols)
